e3nn/o3/tensor_product.py

```diff
@@ -156,6 +162,8 @@ def __init__(
             normalization='component',
             internal_weights=None,
             shared_weights=None,
+            dtype=None,
+            device=None,
```

Comment:
Modules usually dont have `dtype/device` arguments, see for instance [torch.nn.Linear](https://pytorch.org/docs/stable/generated/torch.nn.Linear.html). Why do you want it?

Reply:
Because the original code had this line:
https://github.com/e3nn/e3nn/blob/d279900169466a72470b826cf3a96dc75705fadb/e3nn/o3/tensor_product.py#L595
I think calling `.to()` twice is not efficient, but if convention is more important then I'll rollback those changes.

Reply:
@mariogeiger I'll leave them until your decision or comments, just for sure.

Reply:
ok for adding those arguments

Reply:
Was that original line (595) actually necessary? I'm not sure how anything in the `TensorProduct` would, by default, not be of the default dtype.

Is it the Wigner matrices @mariogeiger ?

Reply:
The original code is long(about 500 lines just for `__init__`), including mostly torchscript generator workarounds. I couldn't figure out the correctness...

Reply:
I have removed all type related keyword arguments in modules, as well as `.to()` call inside `__init__()` function. Now the user should call `.to()` method explicitly, if non-default `torch.dtype` and `torch.device` is required.